<launch>

  <arg name="show_rviz" default="true"/>
  <arg name="broadcast_tf" default="true"/>

  <node name="zed_subscriber" pkg="zed_subscriber" type="zed_subscriber"  output = "screen">
    <!-- <remap from="input_pointcloud" to="/zed2/zed_node/point_cloud/cloud_registered"/> -->
    <remap from="input_zed_transfer" to="/zed_transfer"/>
    <remap from="image_out" to="/zed_image_out/image_rect_color"/>
    <remap from="depth_out" to="/zed_depth_image_out/depth_registered"/>
    <remap from="camera_info_out" to="/zed_depth_image_out/camera_info"/>
  </node>

  <node pkg="nodelet" type="nodelet" name="nodelet_manager" args="manager" />

  <node pkg="nodelet" type="nodelet" name="nodelet1"
        args="load depth_image_proc/point_cloud_xyz nodelet_manager">
    <remap from="camera_info" to="/zed_depth_image_out/camera_info"/>
    <remap from="image_rect" to="/zed_depth_image_out/depth_registered"/>
    <remap from="points" to="/zed_point_cloud_out"/>
  </node> 

  <!--rgb pointcloud-->
  <!--node pkg="nodelet" type="nodelet" name="nodelet1"
        args="load depth_image_proc/point_cloud_xyzrgb nodelet_manager">
    <remap from="rgb/camera_info" to="/zed_depth_image_out/camera_info"/>
    <remap from="rgb/image_rect_color" to="/zed_image_out/image_rect_color"/>
    <remap from="depth_registered/image_rect" to="/zed_depth_image_out/depth_registered"/>
    <remap from="depth_registered/points" to="/zed_point_cloud_out"/>
  </node-->
 
  <node if="$(arg broadcast_tf)" pkg="tf" type="static_transform_publisher" name="zed_broadcaster" args="0 0 0 0 0 0 world zed2_left_camera_frame 100"/>

	<!-- Show RViz -->
	<node if="$(arg show_rviz)" name="zed_publisher_rviz" pkg="rviz" type="rviz" args="-d $(find zed_subscriber)/rviz/sample.rviz"/>

</launch>

